### DISCUSSION TITLE
1.4.0rc3

### DISCUSSION THREAD

**Discussion by eschutho** [apache/superset#17859] [General]
Hello Superset Community,

This is a call for the vote to release Apache Superset version 1.4.0.

The release candidate:
https://dist.apache.org/repos/dist/dev/superset/1.4.0rc3/

Git tag for the release:
https://github.com/apache/superset/tree/1.4.0rc3

The Change Log for the release:
https://github.com/apache/superset/blob/1.4.0rc3/CHANGELOG.md

The Updating instructions for the release:
https://github.com/apache/superset/blob/1.4.0rc3/UPDATING.md

public keys are available at:

https://www.apache.org/dist/superset/KEYS

The vote will be open for at least 72 hours or until the necessary number
of votes are reached.

Please vote accordingly:

[ ] +1 approve
[ ] +0 no opinion
[ ] -1 disapprove with the reason

Thanks,
The Apache Superset Team

<hr /><em>This discussion was created from the release <a href='https://github.com/apache/superset/releases/tag/1.4.0rc3'>1.4.0rc3</a>.</em>

The cherries added to this release candidate on top of rc2 were:
#17781 fix(dashboard): commit update once (@serenajiang)
#17766 fix: Remove positions from json_metadata (@geido)
#17330 fix: import should accept old keys (@betodealmeida)
#17724 ci: temp fix for mysqlclient on an OS regression bug (@dpgaspar)

**Comment by kamalkeshavani-aiinside** [comment]
Requesting to add these PRs in 1.4:
- apache/superset#17657
- apache/superset#17416
- apache/superset#17420
- apache/superset#17615

**Comment by eschutho** [comment]
@kamalkeshavani-aiinside we can only bring in critical bug fixes at this stage of the release. We'll be releasing 1.5.0 next and will be sure to get these new features in. Is the FAB bump a bug fix? If so, I can roll it into 1.4.1.

**Reply by kamalkeshavani-aiinside** [reply]
@eschutho Thanks for the response.
About FAB bump, here is a patch in 3.4.0, which solves the problem of cannot completely sign off with 1 OAUTH provider. So would be great to have it in 1.4.1.
https://github.com/dpgaspar/Flask-AppBuilder/pull/1707

Also, it would be great to know the process of getting PRs in the next release. Say there are a few more which I would like to see in 1.5, so instead of waiting for 1.5rc1 discussion and being pushed to next release, how can I put them up earlier for consideration?
I [tried](https://github.com/apache/superset/pull/17420#issuecomment-973996835) to get them tagged with appropriate tag for next release, but that didn't work out well.

**Reply by eschutho** [reply]
I saw that you mentioned this PR and weren't able to get it tagged via github. You can get early release notifications in the #apache-releases slack channel and also on the dev@ mailing list. Feel free to flag any cherries for the release in the slack channel or vote on the releases. If you report any bugs during an rc1 or rc2 they are usually added in. By later release candidates, the focus shifts toward critical bugs. Thank you for flagging this!

**Reply by eschutho** [reply]
You can also vote for this rc on the mailing list if you feel that this bug is critical and want to see this change made in 1.4.0.

**Reply by kamalkeshavani-aiinside** [reply]
@eschutho I do follow the slack channel, and I did [report](https://github.com/apache/superset/discussions/17528#discussioncomment-1691621) this PR in rc1 as well, but unfortunately that also didn't help. So I will try the other options of putting it in the vote or slack next time. Thank you.

**Reply by eschutho** [reply]
@kamalkeshavani-aiinside apologies for not seeing those cherries in the rc1 discussion. If the vote doesn't pass for rc3, I'll make sure that any bug fixes go into the next rc.

**Comment by jakubczaplicki** [comment]
Is there a chance to add https://github.com/apache/superset/pull/17851 to this release ? Since one of the most recent release (helm chart 0.5.0) we've lost Mapbox map tiles which #17851 hopefully fixes.

**Reply by eschutho** [reply]
Thank you @jakubczaplicki for flagging this PR. I've notated a similar issue in #17923. This PR is for the monorepo, which isn't included in version 1.4, but we'll investigate mapbox issues and see if there's something that we need to pull into superset-ui.

**Comment by eschutho** [comment]
This bug was also reported: https://github.com/apache/superset/issues/17923

**Comment by gnought** [comment]
Thanks for including the FAB 3.4.0 upgrade to fix the logout issue.
Could we also include the other FAB 3.4.3 upgrade to fix roles in Azure OAuth login. The FAB changelog is at here https://github.com/dpgaspar/Flask-AppBuilder/blob/master/CHANGELOG.rst.

PR: #17964

And include the following to close gunicorn security vulnerability please.

PR: #17894

**Reply by eschutho** [reply]
Thank you @gnought for reporting. I have tagged those for 1.4.

**Comment by eschutho** [comment]
The vote for 1.4 rc3 did not pass. We will be adding in the critical bug fixes that have been raised here and in the vote thread. Please continue to report any bugs found in this thread until the rc4 discussion is created. Thank you all!

**Comment by kamalkeshavani-aiinside** [comment]
Thanks for the update on rc3 status.
Here are some more PR requests to be added in 1.4
- apache/superset#17974
- apache/superset#17945
- apache/superset#17726

Also, I would like to see this feature, but not sure if the 2 PRs merged are enough or more work is required.
- apache/superset#14056

**Reply by eschutho** [reply]
Thank you for flagging those fixes. I was able to bring in #17945 but for the other two (#17974 and #17726) are dependent on the monorepo feature that brings Superset-ui into Superset which will be in 1.5.0. The code changes for these bugs are not currently in this repo. Someone will need to make those changes on the old repo and then we need to bump the version on Superset. I'm going to suggest that we move ahead with the one bug fix that we can bring in now, and for 1.4.1 we can add the other two remaining bug fixes into superset-ui. Of course you're encouraged to vote on the release as to whether you feel these fixes are necessary in 1.4.0.

**Reply by eschutho** [reply]
For the alert/report feature, since the feature isn't built yet, we'll need to wait for a future release to get that in. Thank you again for recommending these.

**Reply by kamalkeshavani-aiinside** [reply]
Thanks for the update. I understand the monorepo issue, so I think we can go ahead as you suggested for 1.4.0 and try to get the others in 1.4.1. 
BTW can you also confirm if we can include #17657 and #17416 now? I can see that #17615 is also dependent on monorepo.

**Reply by eschutho** [reply]
Yes, I think getting those bug fixes into 1.4.1 shouldn't be a problem. On the other PRs, since those three are all features, they'll need to go into 1.5.0 as we're only pulling in patch/bug fixes now for 1.4.

**Comment by eschutho** [comment]
A new release candidate is up for voting. I am closing this discussion so that we can move the conversation to the rc4 candidate: https://github.com/apache/superset/discussions/18018 Thank you all for your participation.
